### ISSUE TITLE
CMS [COGNITION]: Multiple ambiguities with linked text in content block

### ISSUE THREAD

**Issue by andaleliz** [department-of-veterans-affairs/va.gov-team#3556]
## Description

Many hub and spoke pages contain a block of text at the bottom of the page containing a heading, followed by descriptive text.   The heading (sometimes an `h4`, sometimes a `b`) and descriptive text are wrapped in an anchor tag.

On hover, the `h4` elements have an underline.  The `b` and `p` elements do not.  On focus, nothing gets an underline.  As a result, these blocks don't follow the WCAG spec that linked text should not rely on color alone, or that links should be visually perceptible. Screenshots below.

https://www.va.gov/health-care/ contains an example of the `b` + `p` mark-up; https://www.va.gov/health-care/about-va-health-benefits/ contains an example of the `h4` + `p` markup.

If this is helpful - @cvalarida identified that this is the chunk in question: https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/site/paragraphs/linkTeaser.drupal.liquid#L27-L41

Side note: the overall pattern is wrapped in a `ul`, and sometimes there's just one `li`.  Is there a better way to handle if there is just one item?

## Acceptance Criteria

As a user, I should be able to visually discern whether or not text is linked.

## Environment

MacBook, Chrome (latest)

## Steps to Recreate

1. Visit https://www.va.gov/health-care/about-va-health-benefits/ and go to the bottom of the page to the section labeled **Related benefits you may qualify for**.  
2. Hover over Disability benefits + text immediately below
3. Repeat w/ keyboard navigation to view results when focus enters that **Related benefits you may qualify for** box

Do the same on https://www.va.gov/health-care/ in the **Other benefits and services** box

## Possible Fixes (optional)

1. Update the markup so only the heading is linked
2. Update the markup so it is consistent across pages (not switching between an `h4` and `b` for the heading.  Is this a definition list candidate? If not, I think an `h4` is appropriate (and feel that it should get an underline all the time for maximum clarity that it is a link)

## WCAG or Vendor Guidance (optional)

1.4.1: Use of Color

## Screenshots or Trace Logs

<img width="912" alt="Screen Shot 2019-11-18 at 3 35 51 PM" src="https://user-images.githubusercontent.com/55992116/69091986-6ac55980-0a19-11ea-826d-98d3a4d0601e.png">
<img width="882" alt="Screen Shot 2019-11-18 at 3 36 29 PM" src="https://user-images.githubusercontent.com/55992116/69091987-6b5df000-0a19-11ea-8476-671c8680161e.png">

**Comment by alexpappasoddball** [comment]
@Karak888 Do you have eyes on this one?

**Comment by Karak888** [comment]
hmm @alexpappasoddball will check with @andaleliz about this one as she is on my team. .:)  thanks!

**Comment by jenstrickland** [comment]
Problematic code:
```
<li {% if parentFieldName === 'field_spokes' %}class="hub-page-link-list__item"{% endif %}>
    {% assign link = linkTeaser.fieldLink | featureSingleValueFieldLink %}
    <a href="{{link.url.path}}" {% if linkTeaser.options["target"] %}target="{{ linkTeaser.options["target"] }}"{% endif %} onClick="recordEvent({ event: 'nav-linkslist' });">
        {% if link.title != empty %}
            <{{ header }} class="{{ headerClass }}">{{ link.title }}</{{ header }}>
            {% if parentFieldName === "field_spokes" %}
                <img class="all-link-arrow" src="/img/arrow-right-blue.svg" alt="right-arrow">
            {% endif %}
        {% endif %}

        {% if linkTeaser.fieldLinkSummary != empty %}
            <p class="va-nav-linkslist-description">{{ linkTeaser.fieldLinkSummary }}</p>
        {% endif %}
    </a>
</li>
```

Fixed code -- move the `</a>` to above the `<p>`:
```
<li {% if parentFieldName === 'field_spokes' %}class="hub-page-link-list__item"{% endif %}>
    {% assign link = linkTeaser.fieldLink | featureSingleValueFieldLink %}
    <a href="{{link.url.path}}" {% if linkTeaser.options["target"] %}target="{{ linkTeaser.options["target"] }}"{% endif %} onClick="recordEvent({ event: 'nav-linkslist' });">
        {% if link.title != empty %}
            <{{ header }} class="{{ headerClass }}">{{ link.title }}</{{ header }}>
            {% if parentFieldName === "field_spokes" %}
                <img class="all-link-arrow" src="/img/arrow-right-blue.svg" alt="right-arrow">
            {% endif %}
        {% endif %}
    </a> 
        {% if linkTeaser.fieldLinkSummary != empty %}
            <p class="va-nav-linkslist-description">{{ linkTeaser.fieldLinkSummary }}</p>
        {% endif %}
</li>
```

**Comment by ncksllvn** [comment]
That's a template used for processing data from Drupal into HTML. If you visit `/health-care`, open your developer console, you'll see `Metalsmith template generated from content:` with that Drupal data. In this case, the data is under the property `fieldRelatedLinks` -

![image](https://user-images.githubusercontent.com/1915775/70738983-d67ea780-1ce3-11ea-8330-d31431d63858.png)

**Comment by jenstrickland** [comment]
Is there a way to adjust the template so that the paragraph is not wrapped in the link, and ideally that the link is inside the heading?

**Comment by ncksllvn** [comment]
That would probably be pretty simple, but IIRC the paragraph being part of the link was a FE requirement as part of the UX. So maybe a designer needs to share insight?

**Comment by jenstrickland** [comment]
From an accessibility point of view, it creates confusion and noise, so if there was a designer that created that it wouldn't have passed an accessibility audit.

**Comment by andaleliz** [comment]
As a designer (although not THE designer who created this pattern) and the person who created this ticket, my insight is that the paragraph should not be part of the link.

**Comment by ncksllvn** [comment]
Okay, I'm not disagreeing - my suggestion for design confirmation was only to avoid the case where FE picks this up for implementation -> the change comes as a surprise to other team members and folks are wondering what's up -> it gets logged as a bug -> FE needs to revisit and do something more. 

@rtwell okay if this ticket proceeds with only linking via the headings, as opposed to the heading + paragraph?

**Comment by rtwell** [comment]
i'm on board to disconnect them @ncksllvn , esp if it brings more clarity. tyvm for everyone's thoughts on this!

**Comment by andaleliz** [comment]
@ncksllvn Excellent point and thanks for looping Ryan in!

**Comment by ncksllvn** [comment]
Thanks @rtwell , sounds like this ticket is ready to go then. I added an estimate and moved to the backlog. I don't think we'll grab it this sprint, but should be ready for next.

**Comment by jenstrickland** [comment]
I have one more point to add: If possible, I would change the structure to be:
```html
<h2>
  <a href="#">Heading link</a>
</h2>
<p>
  Paragraph…
</p>
```

**Comment by ncksllvn** [comment]
Sounds good, thanks @jenstrickland

**Comment by Karak888** [comment]
next action appears to be implementation by FE engineering post discussion in the design system with the team.

**Comment by 1Copenut** [comment]
@jenstrickland && @ncksllvn I'm making a pass through my backlog today and came across this issue. Has this fix been implemented, or is it something that your team opted not to do?

**Comment by 1Copenut** [comment]
@ncksllvn Thanks for the update!

**Comment by meganhkelley** [comment]
@1Copenut @ncksllvn based on your recent comments I'm interpreting this to be within Public Websites' domain, so I've removed my teams label. Please add it back and @ me if that is incorrect. Thanks!

**Comment by 1Copenut** [comment]
Thank you @meganhkelley for the heads up. I was actually looking at a Public Websites new page last night and thought of this issue.

**Comment by 1Copenut** [comment]
@brianalloyd && @andaleliz Let's regroup next week around this item. I think we're on the right track moving the P block out of the complementary content. I'd like to take it a step further by:

* Talk about underlining the links and moving them inside the heading
* Find a way to make the `<aside>` best practice recommendation in axe-core go away

**Comment by brianalloyd** [comment]
@1Copenut thanks for circling back on this issue, @andaleliz is OOO for awhile and can bring in @laurellawrence into the conversation who is backfilling for Liz.  Think it's a great idea to circle up and discuss the details, thanks.

**Comment by 1Copenut** [comment]
Thanks @brianalloyd. Would you mind putting together an invite for myself and @joshkimux to talk with your team? It should be a short discussion, 30 minutes or less I'd think.

**Comment by brianalloyd** [comment]
@1Copenut just sent out an invite, hope the time works as I didn't have full view of calendars.  Look forward to the discussion and thanks again for the support.

**Comment by laurellawrence** [comment]
@brianalloyd and @1Copenut, if you think I could help contribute to resolving this from a design perspective, feel free to add me to the invite

**Comment by brianalloyd** [comment]
@laurellawrence added you, you should have received the invite.  Tax

**Comment by kelsonic** [comment]
Whoa! This is an old one! 🙂

**Comment by brianalloyd** [comment]
@1Copenut if you can update this ticket with a quick summary of what we discussed that will be helpful.  @kelsonic and @ncksllvn this will be something we can pick up next sprint during planning or between initiatives as bandwidth supports.  @laurellawrence feel free to add in any additional details as you feel would be helpful.  Thanks.

**Comment by brianalloyd** [comment]
A quick summation of what needs to be done on this ticket:

- [ ] Move the P block out of the complementary content.
- [ ] Underlining the links and move them inside the heading (should match rest of the styling for the site)
![Screen Shot 2021-01-07 at 3 02 32 PM](https://user-images.githubusercontent.com/70410912/103950195-b7766300-50f9-11eb-847f-a9a6117cc589.png)
_Links are specified above in Ticket description._

**Comment by 1Copenut** [comment]
Thank you @brianalloyd for the comment above. I took a quick look at the `<aside>` content again, and it looks like the P tag has already been moved out of the link block. If we can reverse the order of the H3 and the A tag, and underline the link, we'll be in good shape. 

I **think** this `aside.va-nav-linkslist.va-nav-linkslist--related` is a Liquid template partial, so I hope it can be refactored in one or two places and trickle down. Please let us know if this will be a bigger lift.

Adding a quick screenshot from an exemplar page:

![Screen Shot 2021-01-07 at 4 39 59 PM](https://user-images.githubusercontent.com/934879/103953240-61102100-5107-11eb-9cd3-586ac3df27ea.png)

**Comment by brianalloyd** [comment]
This issue has been resolved, closing out in support based on validation.
